src/components/content/serviceReviews/ApproveOrRejectComments.tsx

```diff
@@ -0,0 +1,117 @@
+    const { TextArea } = Input;
+    const [comments, setComments] = useState<string>('');
+    const approveOrRejectRequest = useApproveOrRejectRequest();
+    const handleOk = () => {
+        if (currentServiceTemplateVo && isApproved !== undefined) {
+            const request: ApproveOrRejectRequestParams = {
+                id: currentServiceTemplateVo.id,
+                reviewRegistrationRequest: {
+                    reviewResult: isApproved
+                        ? ReviewRegistrationRequest.reviewResult.APPROVED
+                        : ReviewRegistrationRequest.reviewResult.REJECTED,
+                    reviewComment: comments,
+                },
+            };
+            approveOrRejectRequest.mutate(request);
+            handleModalClose(true);
+            setComments('');
+        }
+    };
+
+    const handleCancel = () => {
+        setComments('');
+        handleModalClose(true);
+    };
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        setComments(e.target.value);
+    };
+
+    const onClose = () => {
+        getAlertTipCloseStatus(true);
+    };
+
+    if (approveOrRejectRequest.isError) {
+        let errorMessage;
+        if (
+            approveOrRejectRequest.error instanceof ApiError &&
+            approveOrRejectRequest.error.body &&
+            'details' in approveOrRejectRequest.error.body
+        ) {
+            const response: Response = approveOrRejectRequest.error.body as Response;
+            errorMessage = response.details.join();
+        } else {
+            errorMessage = approveOrRejectRequest.error.message;
+        }
+        return (
+            <div className={'approve-reject-alert-class'}>
+                <Alert
+                    message={errorMessage}
+                    description={'Register service review failed.'}
+                    showIcon
+                    closable={true}
+                    onClose={onClose}
+                    type={'error'}
+                />
+            </div>
+        );
+    }
+
+    if (approveOrRejectRequest.isSuccess) {
+        return (
+            <div className={'approve-reject-alert-class'}>
+                <Alert
+                    message={'Processing Status'}
+                    description={'Register service review successfully.'}
+                    showIcon
+                    closable={true}
+                    onClose={onClose}
+                    type={'success'}
+                />
+            </div>
+        );
+    }
+
+    return (
+        <>
+            <Modal
+                title={isApproved !== undefined && isApproved ? 'Approve' : 'Reject'}
+                open={isModalOpen}
+                destroyOnClose={true}
+                onOk={handleOk}
+                onCancel={handleCancel}
+            >
+                <TextArea
+                    rows={10}
+                    placeholder='Please input your comments, max length is 1000.'
```

Comment:
do we need to add the length in the place holder? Can we instead use the `showCount` prop